lte/gateway/c/core/oai/test/amf/test_amf_procedures.cpp

```diff
@@ -1693,4 +1692,38 @@ TEST_F(AMFAppProcedureTest, ServiceRequestSignalling) {
   EXPECT_TRUE(expected_Ids == AMFClientServicer::getInstance().msgtype_stack);
 }
 
-}  // namespace magma5g
\ No newline at end of file
+TEST_F(AMFAppProcedureTest, TestAuthFailureFromSubscribeDbLock) {
+  amf_ue_ngap_id_t ue_id = 0;
+  amf_context_t* amf_ctxt_p = nullptr;
+  nas5g_auth_info_proc_t* auth_info_proc = nullptr;
+  ue_m5gmm_context_s* ue_context_p = nullptr;
+  std::vector<MessagesIds> expected_Ids{
+      AMF_APP_NGAP_AMF_UE_ID_NOTIFICATION,  // new registration notification
+                                            // indication to ngap
+      NGAP_NAS_DL_DATA_REQ,                 // Registration Reject
+      NGAP_UE_CONTEXT_RELEASE_COMMAND       // UEContextReleaseCommand
+  };
+  /* Send the initial UE message */
+  imsi64_t imsi64 = 0;
+  imsi64 = send_initial_ue_message_no_tmsi(amf_app_desc_p, 36, 1, 1, 0, plmn,
+                                           initial_ue_message_hexbuf,
+                                           sizeof(initial_ue_message_hexbuf));
+  /* Check if UE Context is created with correct imsi */
+  EXPECT_TRUE(get_ue_id_from_imsi(amf_app_desc_p, imsi64, &ue_id));
+
+  /* Send the authentication response message from subscriberdb */
+  itti_amf_subs_auth_info_ans_t aia_itti_msg = {};
+  strncpy(aia_itti_msg.imsi, imsi.c_str(), imsi.size());
+  aia_itti_msg.imsi_length = imsi.size();
+  aia_itti_msg.result = DIAMETER_TOO_BUSY;
+  int rc = RETURNerror;
+  rc = amf_nas_proc_authentication_info_answer(&aia_itti_msg);
+  EXPECT_TRUE(rc == RETURNok);
+  ue_context_p = amf_ue_context_exists_amf_ue_ngap_id(ue_id);
```

Comment:
Add NULL check for ue_context_p, amf_ctxt_p and auth_info_proc

Reply:
comment addressed.